packages/headless/src/features/debug/ranking-info-parser.ts

```diff
@@ -1,44 +1,160 @@
-export type RankingInfo = ReturnType<typeof parseRankingInfo>;
+export interface RankingInformation {
+  /**
+   * The attributes of the document that contributed to its ranking.
+   * */
+  documentWeights: DocumentWeights | null;
 
-export interface ListOfWeights {
+  /**
+   * The weight attributed to each term in the query.
+   */
+  termsWeight: TermWeightReport | null;
+
+  /**
+   * The sum of all weights.
+   */
+  totalWeight: number | null;
+
+  /**
+   * The weights applied by query ranking expressions.
+   */
+  qreWeights: QueryRankingExpressionWeights[];
+}
+
+export interface DocumentWeights {
+  /**
+   * The effect of proximity of query terms. More weight is given to documents having the terms closer together.
+   */
   Adjacency: number;
+
+  /**
+   * The weight attributed to user ratings.
+   */
   'Collaborative rating': number;
+
+  /**
+   * The weight assigned through an [indexing pipeline extension (IPE)](https://docs.coveo.com/en/206/) for the item.
+   */
   Custom: number;
+
+  /**
+   * The weight attributed to the document date.
+   */
   Date: number;
+
+  /**
+   * The weight applied by a [query ranking expression (QRE)](https://docs.coveo.com/en/2777/coveo-solutions/using-query-ranking-expressions).
+   */
   QRE: number;
+
+  /**
+   * The proximity of the document in the remaining results after filtering indexed items by query terms and user permissions.
+   * See [item weighting](https://docs.coveo.com/en/1624/searching-with-coveo/understanding-search-result-ranking#phase-2-item-weighting) for more information.
+   */
   Quality: number;
+
+  /**
+   * The weight applied by a [ranking function](https://docs.coveo.com/en/1448/build-a-search-ui/ranking-functions).
+   */
   'Ranking functions': number;
+
+  /**
+   * The effect of the reputation of the document source on the ranking.
+   */
   Source: number;
+
+  /**
+   * The presence of query terms in the document title.
+   */
   Title: number;
+
+  /**
+   * Custom factors affecting the document weight.
+   */
   [key: string]: number;
 }
 
-export interface ListOfQRE {
+export interface QueryRankingExpressionWeights {
+  /**
+   * The query ranking expression (QRE).
+   */
   expression: string;
+
+  /**
+   * The score added by the query ranking expression (QRE).
+   */
   score: number;
 }
 
-export type ListOfTermsWeights = Record<string, WeightsPerTerm>;
+export type TermWeightReport = Record<string, ExpandedTermWeightInformation>;
 
-export interface WeightsPerTerm {
-  Weights: WeightsPerTermBreakdown;
-  terms: Record<string, WeightsPerTermPerDocument>;
+export interface ExpandedTermWeightInformation {
+  Weights: TermWeights | null;
+  terms: Record<string, TermWeightsPerDocument>;
 }
 
-export interface WeightsPerTermBreakdown {
+interface TermWeights {
+  /**
+   * Whether query terms have a special casing in the document.
+   */
   Casing: number;
```

Comment:
Generally, if a property description begins with 'Whether', I'd expect the type of that property to be a boolean. For many of the properties here on `TermWeights`, I'm not sure what the property actually represents. For example, for `Casing`, is it the number of query terms that have special casing?
Same comment on the properties here that begin with 'The presence': okay, but what does the number assigned to the property actually represent?

Reply:
Very good point, these numbers are all weights, not counts per say. I will adjust the wording to better reflect this, something like:

```
The weight allocated when query terms have a special casing in the document.
```